modules/fieldlinessequence/rendering/renderablefieldlinessequence.cpp

```diff
@@ -871,134 +769,107 @@ void RenderableFieldlinesSequence::extractTriggerTimesFromFileNames() {
+        std::string seedFilePath = spFile.path().string();
+        if (!spFile.is_regular_file() || 
+                        seedFilePath.substr(seedFilePath.find_last_of('.')+1) != "txt") {
+            continue;
+        }
+        
+        std::ifstream seedFile(spFile);
+        if (!seedFile.good()) {
+            LERROR(fmt::format("Could not open seed points file '{}'", seedFilePath));
+            outMap.clear();
+            return {};
         }
-    }
-
-    return true;
-}
-
-bool RenderableFieldlinesSequence::extractSeedPointsFromFile(const std::string& path,
-                                                           std::vector<glm::vec3>& outVec)
-{
 
-    std::ifstream seedFile(path);
-    if (!seedFile.good()) {
-        LERROR(fmt::format("Could not open seed points file '{}'", path));
-        return false;
-    }
+        LDEBUG(fmt::format("Reading seed points from file '{}'", seedFilePath));
+        std::string line;
+        std::vector<glm::vec3> outVec;
+        while (std::getline(seedFile, line)) {
+            std::stringstream ss(line);
+            glm::vec3 point;
+            ss >> point.x;
+            ss >> point.y;
+            ss >> point.z;
+            outVec.push_back(std::move(point));
+        }
 
-    LDEBUG(fmt::format("Reading seed points from file '{}'", path));
-    std::string line;
-    while (std::getline(seedFile, line)) {
-        std::stringstream ss(line);
-        glm::vec3 point;
-        ss >> point.x;
-        ss >> point.y;
-        ss >> point.z;
-        outVec.push_back(std::move(point));
-    }
+        if (outVec.size() == 0) {
+            LERROR(fmt::format("Found no seed points in: {}", seedFilePath));
+            outMap.clear();
+            return {};
+        }
 
-    if (outVec.size() == 0) {
-        LERROR(fmt::format("Found no seed points in: {}", path));
-        return false;
+        size_t lastIndex = seedFilePath.find_last_of('.');
+        std::string name = seedFilePath.substr(0, lastIndex);   // remove file extention
+        size_t dateAndTimeSeperator = name.find_last_of('_');
+        std::string time = name.substr(dateAndTimeSeperator + 1, name.length());
+        std::string date = name.substr(dateAndTimeSeperator - 8, 8);    //8 for yyyymmdd
+        std::string dateAndTime = date + time;
+            
+        // add outVec as value and time stamp as int as key
+        outMap[dateAndTime] = outVec;
     }
-
-    return true;
+    return outMap;
 }
 
-void RenderableFieldlinesSequence::extractMagnitudeVarsFromStrings(
-                                                      std::vector<std::string>& extraVars,
-                                                   std::vector<std::string>& extraMagVars)
-{
-
-    for (int i = 0; i < static_cast<int>(extraVars.size()); i++) {
-        const std::string& str = extraVars[i];
+std::vector<std::string> RenderableFieldlinesSequence::extractMagnitudeVarsFromStrings() {
```

Comment:
Passing in the `_extraVars` as a parameter means this function doesn't have to be defined inside the class anymore